daemon/src/maker_inc_connections.rs

```diff
@@ -156,57 +160,50 @@ impl Actor {
 impl Actor {
     async fn handle_broadcast_order(&mut self, msg: BroadcastOrder) -> Result<()> {
         let order = msg.0;
-        for (taker_id, conn) in self.write_connections.clone() {
-            if conn
-                .do_send_async(wire::MakerToTaker::CurrentOrder(order.clone()))
-                .await
-                .is_err()
-            {
-                tracing::trace!(%taker_id, "removing outdated connection to taker because unable to send order: {:?}", order);
-                self.write_connections.remove(&taker_id);
-            } else {
-                tracing::trace!(%taker_id, "sent new order: {:?}", order);
-            }
+        for taker_id in self.write_connections.clone().keys() {
+            self.send_to_taker(taker_id, wire::MakerToTaker::CurrentOrder(order.clone())).await.expect("send_to_taker only fails on missing hashmap entry and we are iterating over those entries");
```

Comment:
❓ Can it happen that we access the hashmap from a different handle block and an element gets removed while we are looping in here? 

It would be great if we can remove the failure scenario from the for loop. Regardless of "it cannot happen". I'm afraid that changing code might re-introduce problems that we already solved. I think we should not fail in the loop no matter what.

Reply:
We are holding a mutable reference to `Actor` here. These kind of race conditions are impossible in Rust.

Reply:
If this panics, someone modified the `send_to_taker` function without thinking through all call-sites which is a bug and should be fixed.

Reply:
I still think it is bad to have a failure scenario in a loop and I don't think we should expect this, because if `send_to_taker` changes we might re-introduce an error that was hard to track down.